### PR TITLE
Optionally set the correct_dataset from ws_code.string_value

### PR DESCRIPTION
**Pull request by as2388**
Copied from https://github.com/AfricasVoices/Project-RVI-Elections/pull/9/files, but put behind an optional flag this time so we don't use this behaviour by accident.

### CODE REVIEW

File: src/engagement_db_coda_sync/lib.py [hunk 1]
```diff
@@ -287,15 +287,25 @@ def _update_engagement_db_message_from_coda_message(engagement_db, engagement_db
 
     # WS-correct if there is a valid ws_code
     if ws_code is not None:
+        # Establish the correct dataset to move this message to.
+        # To determine the dataset, the following strategies are tried, in order:
+        #  1. Search the other dataset configurations for a match.
+        #  2. If `set_dataset_from_ws_string_value` has been set, move the message to the dataset `ws_code.string_value`
+        #  3. If the `default_ws_dataset` has been specified, move the message to this default dataset.
+        #  4. Raise a ValueError.
```
Comment (DanielMwendwa) [inline src/engagement_db_coda_sync/lib.py:295]: Should there be an else clause like - "in case we can't find the correct dataset from the config then...`

**Review by DanielMwendwa — APPROVED** [review]

File: src/engagement_db_coda_sync/lib.py [hunk 2]
```diff
@@ -287,15 +287,25 @@ def _update_engagement_db_message_from_coda_message(engagement_db, engagement_db
...
         try:
             correct_dataset = \
                 coda_config.get_dataset_config_by_ws_code_string_value(ws_code.string_value).engagement_db_dataset
```
[the remark this answers is not in the document]
Comment (as2388) [reply src/engagement_db_coda_sync/lib.py:298]: yes, let me do in a separate PR

File: src/engagement_db_coda_sync/lib.py (hunk 1 above)
Comment (as2388) [reply src/engagement_db_coda_sync/lib.py:295]: 4. was implicitly the else clause - but I'll edit to make it more obvious.

Comment (as2388) [reply src/engagement_db_coda_sync/lib.py]: I've adjusted this to try to make it clearer

**Review by IsaackMwenda — APPROVED** [review]

**Pull request merged**